model/buildlogger_storage.go

```diff
@@ -0,0 +1,67 @@
+package model
+
+import (
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/mongodb/anser/bsonutil"
+	"github.com/pkg/errors"
+)
+
+// S3 Permissions is a type that describes the object canned ACL from S3.
+type S3Permissions string
+
+const (
+	S3PermissionsPrivate                S3Permissions = s3.ObjectCannedACLPrivate
+	S3PermissionsPublicRead             S3Permissions = s3.ObjectCannedACLPublicRead
+	S3PermissionsPublicReadWrite        S3Permissions = s3.ObjectCannedACLPublicReadWrite
+	S3PermissionsAuthenticatedRead      S3Permissions = s3.ObjectCannedACLAuthenticatedRead
+	S3PermissionsAWSExecRead            S3Permissions = s3.ObjectCannedACLAwsExecRead
+	S3PermissionsBucketOwnerRead        S3Permissions = s3.ObjectCannedACLBucketOwnerRead
+	S3PermissionsBucketOwnerFullControl S3Permissions = s3.ObjectCannedACLBucketOwnerFullControl
```

Comment:
should these be bucket permissions rather than make it s3 specific? and avoid the dependency on the s3 pacakge?

Reply:
so move these to pail? I was thinking that but we already depend on the aws-sdk-go package in cost package.

Reply:
I think this should be in pail, and refer to a concept/abstraction/etc that isn't s3 specific.